### DISCUSSION TITLE
Enable/Disable features

### DISCUSSION THREAD

**Discussion by mtnsec** [Security-Onion-Solutions/securityonion#2434] [Unsupported Versions]
During installation of my manager/search server, I unselected several options because of resource usage warnings. I'd like to start enabling these one at a time to identify the resource impact and evaluate whether I can keep them enabled. I'm unable to find instructions in the documentation for how to do this.

How can I enable/install individual components? And related - if any of them do cause too much resource utilization - how do I uninstall/disable them?

Thanks,

**Comment by 0nnyx** [comment]
That might help https://github.com/Security-Onion-Solutions/securityonion/discussions/2009

**Comment by mtnsec** [comment]
Thanks!

Unfortunately it didn't help.  I started with trying to enable Fleet - I just flipped the False to True but there are a lot of other configs in global.sls I wasn't sure what to do with.  I got some errors about TLS and mysql. They're on a different system so I'll have to try setting the other configs, and if it doesn't work I'll come back with actual logs.

**Reply by 0nnyx** [reply]
Sorry mate, that method worked fine for me to enable wazuh but I had already osquery enabled initially.

**Reply by mtnsec** [reply]
I appreciate the nudge in the right direction anyways!  Hopefully I just messed up my configs somehow. :)

**Comment by mtnsec** [comment]
In my global.sls I set (not sure what the timestamp format is supposed to be so I left it alone)

```
fleet_custom_hostname: 'myhostname'
  fleet_manager: True
  fleet_node: True
  fleet_packages-timestamp: 'N/A'
  fleet_packages-version: 1
  fleet_hostname: 'myhostname'
  fleet_ip: '#.#.#.#'
```
Here are the errors

```
$ sudo salt-call state.apply fleet
[WARNING ] /usr/lib/python3/dist-packages/salt/modules/mysql.py:656: Warning: (1681, "'PASSWORD' is deprecated and will be removed in a future release.")
  return cur.execute(qry, args)
```

```
$ sudo salt-call state.highstate
[WARNING ] State for file: /etc/pki/ca.crt - Neither 'source' nor 'contents' nor 'contents_pillar' nor 'contents_grains' was defined, yet 'replace' was set to 'True'. As there is no source to replace the file with, 'replace' has been set to 'False' to avoid reading the file unnecessarily.
[WARNING ] State for file: /etc/pki/ca.crt - Neither 'source' nor 'contents' nor 'contents_pillar' nor 'contents_grains' was defined, yet 'replace' was set to 'True'. As there is no source to replace the file with, 'replace' has been set to 'False' to avoid reading the file unnecessarily.
```

When I try to navigate to fleet, I still get a 404 error as well - which is the same as before trying to enable it so no change there.

(Editing to try to fix the formatting...)

**Reply by weslambert** [reply]
@defensivedepth any thoughts with regard to Osquery/Fleet initialization after install?

**Reply by defensivedepth** [reply]
It is not something that is currently supported, but will be eventually.
